e2e/app/element/base-element.ts

```diff
@@ -167,6 +167,9 @@ export default class BaseElement {
         await element.hover();
         await this.page.waitForTimeout(500);
       }
+      if (previousX === undefined || previousY === undefined) {
+        throw new Error(`Failed to click element because the it is not visible in the viewport. (${this.getXpath()})`);
+      }
```

Comment:
Click will not work if the element was not visible. Throws error to end the test here.